std/file.d

```diff
@@ -4458,11 +4458,16 @@ void rmdirRecurse(ref DirEntry de)
     }
     else
     {
-        // all children, recursively depth-first
-        foreach (DirEntry e; dirEntries(de.name, SpanMode.depth, false))
-        {
-            attrIsDir(e.linkAttributes) ? rmdir(e.name) : remove(e.name);
-        }
+        // dirEntries is @system because it uses a DirIterator with a
+        // RefCounted variable, but here, no references to the payload is
+        // escaped to the outside, so this should be @trusted
```

Comment:
Can't this be verified by the compiler by making the `ref DirEntry de` `scope`?

Reply:
No because `.dirEntries` is @system and `scope` will do nothing about safety here. RefCounted is @system because the implementation uses malloc/free.

Reply:
See this interesting discussion about @safe ref counting on dlang forum: https://forum.dlang.org/thread/r8ilpu$2mk7$1@digitalmars.com